src/System.Drawing.Common/tests/Printing/PaperSizeTest.cs

```diff
@@ -0,0 +1,167 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.using System;
+//
+// Copyright (C) 2009 Novell, Inc (http://www.novell.com)
+//
+// Permission is hereby granted, free of charge, to any person obtaining
+// a copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+//
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+// Author:
+// 	Andy Hume <andyhume32@yahoo.co.uk>
+//
+
+using Xunit;
+
+namespace System.Drawing.Printing.Test
+{
+    public class PaperSizeTest
+    {
+
+        [Fact]
+        public void ChangeHeight_OnCustomPaperKind_ShouldNotThrowAnExcetion()
+        {
+            PaperSize ps = new PaperSize("foo", 100, 100);
+
+            ps.Height = 1;
+
+            Assert.Equal(1, ps.Height);
+        }
+
+        [Fact]
+        public void ChangeHeight_OnFixedPaperKind_ShouldThrowAnException()
+        {
+            PaperSize ps = new PaperSize("foo", 100, 100);
+
+            ps.RawKind = (int)PaperKind.A4;
+
+            Assert.Throws<ArgumentException>(() => ps.Height = 2);
+        }
+
+        [Fact]
+        public void ChangeHeight_OnNotRealCustomKind_ShouldThrowAnException()
+        {
+            PaperSize ps = new PaperSize("foo", 100, 100);
+
+            ps.RawKind = 999999;
+
+            // The properties can be changed only when the *real* Kind is Custom 
+            // and not when is 'effectively' Custom.
+            Assert.Throws<ArgumentException>(() => ps.Height = 4);
+        }
+
+        [Fact]
+        public void ChangeHeight_OnCustomKind_ShouldNotThrowAnException()
+        {
+            PaperSize ps = new PaperSize("foo", 100, 100);
+
```

Comment:
nit: Remove extra spaces. I'll stop commenting about this one, just try to clean up any extra spaces where they don't really help clarity.